frontend/common/src/hooks/useCustomEvent.ts

```diff
@@ -0,0 +1,23 @@
+import { ReactPlugin } from "@microsoft/applicationinsights-react-js";
+import { ICustomProperties } from "@microsoft/applicationinsights-web";
+import { useState, useEffect, useRef } from "react";
+
+export default function useCustomEvent(
```

Comment:
Could we give this a more descriptive name?  Maybe useAppInsightsEvent?

Reply:
hmm ..all the events triggered by this type will show up as "custom event" in App Insights. It is the type of the event.

Reply:
Maybe useAppInsightsCustomEvent then?  It would make it clear that it is associated with then other AppInsights components.  I'm worried that useCustomEvent is too generic to have any meaning.

Reply:
Uh oh, typo:
![image](https://user-images.githubusercontent.com/8978655/210875865-94fad9c1-1d8c-4b1d-a101-d0e700a79417.png)